Add support for password widget

lucor opened the pull request:
This PR extends the text editor adding support for contents are visibly obfuscated like passwords.

aarzilli commented:
This is too complicated. Especially the part where there are now two buffers (Buffer and password) and they need some duplicated handling. I think it would be better to alter the way the input field is drawn by either drawing nothing or drawing a bunch of dots if the EditPassword flag is set.

lucor commented:
Agreed. I've updated the code to avoid additional buffer and added logic into the Draw receiver.
Right now it makes text transparent, but it'd great to have something like `*` displayed. 
Tried to drawn the char but it is stored into buffer the on the next `updatefn` refresh making it not usable.
Wondering if a font password is the way to go or if you have other suggestions.

aarzilli commented:
This is closer but I prefer to go with something more like this: 280c1bac14a54399b20ed05657751272fa0db129.
Thank you.
